### DISCUSSION TITLE
create-callback

### DISCUSSION THREAD

**Discussion by MauricioOlitel** [twilio-professional-services/flex-project-template#612] [General]
I'm trying to use the callback to create a function within a WhatsApp service, so the customer can choose to change the service channel, I tested the callback on a voice call and it worked great, but when I add the function to a flow from whatsapp I get the error below:
Unauthorized: Token was not provided.
I tried to call the function through postmam and got the same error, is there a specific parameter that I have to send?

**Comment by dremin** [comment]
Hi @MauricioOlitel! If the function you created is wrapped by `prepareFlexFunction()`, it requires a token from a logged-in Flex worker. Since this function won't be called by the worker's Flex UI, you can instead wrap the function with `prepareStudioFunction()` which does not have this requirement.

**Comment by MauricioOlitel** [comment]
ok, thank you very much
